### PR TITLE
Update sample app

### PR DESCRIPTION
**Pull request by aenand**
Updating the sample app to use the Spreedly test provider and include exemption types. Testing instructions posted in slack.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ var logger = require('morgan');
 
 var indexRouter = require('./routes/index');
 
-var port = process.env.PORT || 8080;
```
Comment (tatsianaclifton) [inline app.js:11]: Why do you need to change a port?

File: public/javascripts/your-code.js [hunk 2]
```diff
@@ -68,9 +68,17 @@ function onPaymentMethodFn(token, paymentMethodData) {
...
           updateUI({
-            token: data.token
+            token: data.token,
+            message: data.message
           }, true);
           break;
+        case "failed":
+            // Authentication failed
+            updateUI({
+              token: data.token,
```
Comment (tatsianaclifton) [inline public/javascripts/your-code.js:78]: I didn't see token in UI on failure
Screen Shot 2022-05-12 at 10 47 13 AM

File: routes/index.js [hunk 3]
```diff
@@ -69,7 +79,10 @@ router.post('/attempt-purchase', function (req, res, next) {
       browser_info: frontendTx.browserData,
       redirect_url: REDIRECT_URL,
       callback_url: CALLBACK_URL,
-      //sca_provider_key: `${SCA_PROVIDER_KEY}`
+      sca_provider_key: `${SCA_PROVIDER_KEY}`,
+      sca_authentication_parameters: 
+        TEST_SCENARIOS.get(String(frontendTx.amount))
+
```
Comment (tatsianaclifton) [inline routes/index.js]: extra line

**Review by tatsianaclifton — COMMENTED** [review]
Overall looks very good. Couple issues I noticed:
- some inline comments
- app crashed for me when making payment with 50.00 euro
- when failing the challenge the error says undefiled 
Screen Shot 2022-05-12 at 10 56 16 AM

File: public/javascripts/your-code.js (hunk 2 above)
Comment (aenand) [reply public/javascripts/your-code.js:78]: Added a fix that displays token + message or context

File: app.js (hunk 1 above)
Comment (aenand) [reply app.js:11]: CRDB runs on port 8080 so when I was testing I couldn't start the server. I can drop this change

**Review by tatsianaclifton — APPROVED** [review]
Good work!

**Pull request merged**